maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/extensions/AuthenticationException.java

```diff
@@ -0,0 +1,39 @@
+package org.apache.maven.plugin.surefire.extensions;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.maven.surefire.extensions.ForkChannel;
+import org.apache.maven.surefire.extensions.util.CommandlineExecutor;
+
+import java.io.IOException;
+
+/**
+ * After the authentication has failed, {@link ForkChannel#connectToClient()} throws {@link AuthenticationException}
+ * and {@link org.apache.maven.plugin.surefire.booterclient.ForkStarter} should close {@link CommandlineExecutor}.
+ *
+ * @since 3.0.0-M5
+ */
+public class AuthenticationException extends IOException
```

Comment:
I think since we've got rid the `auth` term, but this class name still refers to it. What about `InvalidSessionIdEx` extends IllegalArgumentEx`. This is clearly not an `IOEx`.

Reply:
Sorry but the callsequences expect IOException upto ForkStarter. I could simply throw the IOException but the mesage might be lost or the users do not read them and the stack trace with authentication exception would be understood by every developer.

Reply:
OK, let's accept that limitation, but use a proper name: `InvalidSessionIdEx  extends IOEx`.